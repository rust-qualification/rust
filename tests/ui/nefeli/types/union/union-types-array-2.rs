// Union Types: d1b5850a-f09d-4785-9d56-6ec53d7cfccf

fn main() {
    union InvalidArrayUnion {
        x: [String; 3], //~ ERROR field must implement `Copy` or be wrapped in `ManuallyDrop<...>` to be used in a union
    }
}