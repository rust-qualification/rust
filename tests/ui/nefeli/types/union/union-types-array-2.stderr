error[E0740]: field must implement `Copy` or be wrapped in `ManuallyDrop<...>` to be used in a union
  --> $DIR/union-types-array-2.rs:5:9
   |
LL |         x: [String; 3],
   |         ^^^^^^^^^^^^^^
   |
   = note: union fields must not have drop side-effects, which is currently enforced via either `Copy` or `ManuallyDrop<...>`
help: wrap the field type in `ManuallyDrop<...>`
   |
LL |         x: std::mem::ManuallyDrop<[String; 3]>,
   |            +++++++++++++++++++++++           +

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0740`.
